llvm/test/DebugInfo/X86/basic-block-sections_1.ll

```diff
@@ -16,10 +16,10 @@
 ; NO-SECTIONS: DW_AT_high_pc [DW_FORM_data4] ({{.*}})
 ; BB-SECTIONS: DW_AT_low_pc [DW_FORM_addr] (0x0000000000000000)
 ; BB-SECTIONS-NEXT: DW_AT_ranges [DW_FORM_sec_offset]
+; BB-SECTIONS-NEXT: [{{.*}}) ".text.hot._Z3fooi"
```

Comment:
Why did the order change here?

Reply:
In [here](https://github.com/llvm/llvm-project/blob/488640389faedee8c65566275441779faa620fd2/llvm/lib/CodeGen/AsmPrinter/DwarfCompileUnit.cpp#L688),

MBBSectionRanges is used to populate DW_AT_ranges.  MBBSectionRanges is a MapVector with deterministic order and I just changed that order to have the entry block's section as the first section in AsmPrinter.cpp.